mypyc/irbuild/ll_builder.py

```diff
@@ -622,6 +624,24 @@ def compare_tagged(self, lhs: Value, rhs: Value, op: str, line: int) -> Value:
         self.goto_and_activate(out)
         return result
 
+    def compare_strings(self, lhs: Value, rhs: Value, op: str, line: int) -> Value:
+        """Compare two strings"""
+        compare_result = self.call_c(unicode_comapre, [lhs, rhs], line)
+        error_constant = self.add(LoadInt(-1, line, c_int_rprimitive))
+        compare_error_check = self.add(ComparisonOp(compare_result,
+                                                    error_constant, ComparisonOp.EQ, line))
+        exception_check, final_compare = BasicBlock(), BasicBlock()
+        branch = Branch(compare_error_check, exception_check, final_compare, Branch.BOOL_EXPR)
+        branch.negated = False
+        self.add(branch)
+        self.activate_block(exception_check)
+        self.call_c(no_err_occurred_op, [], line)
```

Comment:
Call `PyErr_Occurred` directly to avoid an extra function call in case the function is not inlined in a non-C back end?

Reply:
I am not sure what does "call `PyErr_Occurred`" mean in your comment. The no_err_occurred_op here, if you look at its definition, it's just calling the function and checking the return value.

Reply:
The problem with directly calling `PyErr_Occurred` is that it returns `NULL` on normal exit and non-NULL on error, while `ERR_MAGIC` handles the exception when `NULL`, which means we'd add extra logic to negate the check on this single op.

Reply:
Maybe you can can call `PyErr_Occurred`, compare the return against NULL (`ComparisonOp`), and if it's NULL, use `CPy_KeepPropagating`?

Reply:
That should work. Let me do a quick update.